File: 2026_GFG_POTD/FEBRUARY/09-02-2026/Find Kth Rotation.java
class Solution {
    
    public int binarySearch(int[] arr){
        int left = 0;
        int right = arr.length - 1;
        int index = -1;
        int n = arr.length;
        
        while(left <= right){
            int mid = left + (right - left) / 2;
            if(arr[mid] <= arr[n - 1]){
                index = mid;
                right = mid - 1;
            }else{
                left = mid + 1;
            }
        }
        return index;
    }
    public int findKRotation(int arr[]) {
        // Code here
        return binarySearch(arr);
    }
}

File: 2026_GFG_POTD/FEBRUARY/09-02-2026/Find Kth Rotation.py
class Solution:
    def findKRotation(self, arr):
        # code here
        if arr[0] <= arr[-1]:
            return 0
        lo, hi = 0, len(arr)
        while lo < hi:
            mi = lo + (hi-lo)//2
            if arr[mi] < arr[0]:
                hi = mi
            else:
                lo = mi+1
        return lo